src/screens/Leaderboard/Leaderboard.tsx

```diff
@@ -1,129 +1,127 @@
 import './Leaderboard.css';
-import { Button, Spinner, Title } from '@tg-app/ui';
-import { useEffect, useMemo, useState } from 'react';
-import { Score, ScoreProps } from '~/components/Leaderboard/Score/Score.tsx';
+import { Button, Spinner } from '@tg-app/ui';
+import { useEffect, useState } from 'react';
 import { AnalyticsId } from '@tg-app/analytics';
 import { ActiveTab } from '~/App.tsx';
-import { leaderboardDataMock } from '~/components/Leaderboard/leaderboard-mock.ts';
-import { useEvents, useWallet } from '~/hooks';
+import { useBot, useEvents, useWallet } from '~/hooks';
 import { ActivityEvent } from '@cere-activity-sdk/events';
+import { EngagementEventData } from '~/screens';
+import * as hbs from 'handlebars';
 import { EVENT_APP_ID } from '~/constants.ts';
 
 type LeaderboardProps = {
   setActiveTab: (tab: ActiveTab) => void;
 };
+
+hbs.registerHelper('json', (context) => JSON.stringify(context));
+
 export const Leaderboard = ({ setActiveTab }: LeaderboardProps) => {
-  const [leaderboardData, setLeaderboardData] = useState<Omit<ScoreProps, 'rank'>[]>([]);
-  const [isLoading, setLoading] = useState(false);
+  const [leaderboardHtml, setLeaderboardHtml] = useState<string>('');
+  const [isLoading, setLoading] = useState(true);
 
   const { account } = useWallet();
+  const bot = useBot();
   const eventSource = useEvents();
 
-  const userScore: ScoreProps | undefined = useMemo(
-    () =>
-      leaderboardData
-        .sort((a, b) => b.score - a.score)
-        .reduce(
-          (acc, score, index) => {
-            if (acc) return acc;
-            // if (score.user === '0x002...') {
-            if (score.user === account?.address) {
-              return {
-                ...score,
-                rank: index + 1,
-              };
-            }
-            return undefined;
-          },
-          undefined as ScoreProps | undefined,
-        ),
-    [account?.address, leaderboardData],
-  );
-
   useEffect(() => {
-    setLoading(true);
+    const fetchData = async () => {
+      try {
+        setLoading(true);
 
-    eventSource.isReady().then(
-      (ready) => {
+        const ready = await eventSource.isReady();
         console.log('EventSource ready:', ready);
 
         const { event_type, timestamp, userPubKey, appPubKey, data } = {
           event_type: 'GET_LEADERBOARD',
           timestamp: '2024-11-15T09:01:01Z',
-          userPubKey: '2ce686f936c69f91d91c30b4f4c6dc54d20dc13e50cdfba0b98f63dc57f27b78',
-          appPubKey: '2102',
+          userPubKey: account?.publicKey,
+          appPubKey: EVENT_APP_ID,
```

Comment:
I don't think appId is appPubKey but we can keep it this way for now until we apply ROB's new changes here. I think we have implemented real appPublicKey there, right?